airflow/models/connection.py

```diff
@@ -43,6 +47,26 @@ def parse_netloc_to_hostname(*args, **kwargs):
     return _parse_netloc_to_hostname(*args, **kwargs)
 
 
+def sanitize_conn_id(conn_id: str | None) -> str | None:
+    """
+    Sanitises the connection id and allows only specific characters to be within. Namely,
+    it allows alphanumeric characters plus the symbols @,#,$,%,&,!,-,_, and () from 1
+    and up to 200 consecutive matches.
+
+    The character selection is such that it prevents the injection of javascript or
+    executable bits in order to avoid any awkward behavior in the front-end.
+
+    :param conn_id: The connection id to sanitize.
+    :return: the sanitized string, `None` otherwise.
```

Comment:
Can we return the sanitised string or FAIL instead?
cc @vincbeck @uranusjr

Reply:
I did what the initial behaviour was - if the sanitized string is `None` then an exception is raised by SQL Alchemy anyway, as this is a non-NULLable field.